### PR TITLE
Add Endpoint add/delete methods for tenant levels

### PR DESCRIPTION
**Pull request by Akila94**

### CODE REVIEW

File: components/mediation-admin/org.wso2.carbon.endpoint/src/main/java/org/wso2/carbon/endpoint/service/EndpointAdmin.java [hunk 1]
```diff
@@ -198,6 +199,49 @@ public boolean addEndpoint(String epString) throws EndpointAdminException {
...
+     * @param tenantDomain Domain of the logged tenant
+     * @return
+     * @throws EndpointAdminException Thrown if an error occur
+     */
+    public boolean deleteEndpointForTenant(String endpointName, String tenantDomain) throws EndpointAdminException {
+        try {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
+            boolean status = deleteEndpoint(endpointName);
+            return status;
```
Comment (milanperera) [inline components/mediation-admin/org.wso2.carbon.endpoint/src/main/java/org/wso2/carbon/endpoint/service/EndpointAdmin.java]: we can remove the status variable by calling "return deleteEndpoint(endpointName);"

File: components/mediation-admin/org.wso2.carbon.endpoint/src/main/java/org/wso2/carbon/endpoint/service/EndpointAdmin.java [hunk 2]
```diff
@@ -198,6 +199,49 @@ public boolean addEndpoint(String epString) throws EndpointAdminException {
...
+     * @return
+     * @throws EndpointAdminException Thrown if an error occur
+     */
+    public boolean addEndpointForTenant(String endpointData, String tenantDomain) throws EndpointAdminException {
+        try {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain,
+                    true);
+            boolean status = addEndpoint(endpointData);
+            return status;
```
Comment (milanperera) [inline components/mediation-admin/org.wso2.carbon.endpoint/src/main/java/org/wso2/carbon/endpoint/service/EndpointAdmin.java]: we can remove the status variable by calling "return addEndpoint(endpointData);"

File: components/mediation-admin/org.wso2.carbon.endpoint/src/main/java/org/wso2/carbon/endpoint/service/EndpointAdmin.java [hunk 3]
```diff
@@ -198,6 +199,49 @@ public boolean addEndpoint(String epString) throws EndpointAdminException {
...
+     * @param endpointData Endpoint file content
+     * @param tenantDomain Domain of the logged tenant
+     * @return
+     * @throws EndpointAdminException Thrown if an error occur
+     */
+    public boolean addEndpointForTenant(String endpointData, String tenantDomain) throws EndpointAdminException {
+        try {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain,
+                    true);
```
Comment (milanperera) [inline components/mediation-admin/org.wso2.carbon.endpoint/src/main/java/org/wso2/carbon/endpoint/service/EndpointAdmin.java]: shall we properly format this?

File: components/mediation-admin/org.wso2.carbon.endpoint/src/main/java/org/wso2/carbon/endpoint/service/EndpointAdmin.java [hunk 4]
```diff
@@ -198,6 +199,49 @@ public boolean addEndpoint(String epString) throws EndpointAdminException {
         return false;
     }
 
+    /**
+     * Set the tenant domain when a publisher publishes his API in MT mode. When publisher publishes
+     * the API, we login the gateway as supretenant. But we need to create the endpoint file in the
```
Comment (vinok88) [inline components/mediation-admin/org.wso2.carbon.endpoint/src/main/java/org/wso2/carbon/endpoint/service/EndpointAdmin.java]: Typo in the word "supretenant"

Comment (Akila94) [reply components/mediation-admin/org.wso2.carbon.endpoint/src/main/java/org/wso2/carbon/endpoint/service/EndpointAdmin.java]: Corrected

**Comment by Akila94** [comment]
Added changes to wsdl

File: components/mediation-admin/org.wso2.carbon.endpoint/src/main/java/org/wso2/carbon/endpoint/service/EndpointAdmin.java (hunk 3 above)
Comment (Akila94) [reply components/mediation-admin/org.wso2.carbon.endpoint/src/main/java/org/wso2/carbon/endpoint/service/EndpointAdmin.java]: Done

**Comment by Akila94** [comment]
Added changes

**Pull request merged**